src/db/migration/v0_19_0.rs

```diff
@@ -0,0 +1,340 @@
+// Copyright 2019-2024 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+//! Migration logic for 0.18.0 to 0.19.0 version.
+//! For the need for Ethereum RPC API, a new column in parity-db has been introduced to handle
+//! mapping of values such as, `Hash` to `TipsetKey` and `Hash` to message `Cid`.
+
+use crate::chain::ChainStore;
+use crate::cli_shared::chain_path;
+use crate::daemon::db_util::{cache_tipset_keys, load_all_forest_cars};
+use crate::db::car::ManyCar;
+use crate::db::db_engine::{open_db, Db};
+use crate::db::migration::migration_map::temporary_db_name;
+use crate::db::migration::v0_19_0::paritydb_0_18_0::{DbColumn, ParityDb};
+use crate::genesis::read_genesis_header;
+use crate::networks::ChainConfig;
+use crate::state_manager::StateManager;
+use crate::Config;
+use anyhow::Context;
+use cid::multihash::Code::Blake2b256;
+use cid::multihash::MultihashDigest;
+use cid::Cid;
+use fs_extra::dir::CopyOptions;
+use fvm_ipld_encoding::DAG_CBOR;
+use semver::Version;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use strum::IntoEnumIterator;
+use tracing::info;
+
+use super::migration_map::MigrationOperation;
+
+pub(super) struct Migration0_18_0_0_19_0 {
+    from: Version,
+    to: Version,
+}
+
+/// Migrates the database from version 0.18.0 to 0.19.0
+impl MigrationOperation for Migration0_18_0_0_19_0 {
+    fn new(from: Version, to: Version) -> Self
+    where
+        Self: Sized,
+    {
+        Self { from, to }
+    }
+
+    fn pre_checks(&self, _chain_data_path: &Path) -> anyhow::Result<()> {
+        Ok(())
+    }
+
+    fn migrate(&self, chain_data_path: &Path, config: &Config) -> anyhow::Result<PathBuf> {
+        let source_db = chain_data_path.join(self.from.to_string());
+
+        let temp_db_path = chain_data_path.join(temporary_db_name(&self.from, &self.to));
+        if temp_db_path.exists() {
+            info!(
+                "Removing old temporary database {temp_db_path}",
+                temp_db_path = temp_db_path.display()
```

Comment:
same info in two different ways?

Reply:
Indeed, we could make use of some service functions here. Let's keep the refactor for a subsequent PR.